src/sentry/discover/models.py

```diff
@@ -49,6 +67,7 @@ class DiscoverSavedQuery(Model):
     visits = BoundedBigIntegerField(null=True, default=1)
     last_visited = models.DateTimeField(null=True, default=timezone.now)
     is_homepage = models.BooleanField(null=True, blank=True)
+    dataset = BoundedPositiveIntegerField(choices=DiscoverSavedQueryTypes.as_choices(), null=True)
```

Comment:
Do we want to have a default here, or attempt to backfill? If all existing rows are of one type it could make sense

Reply:
When I did this for dashboards, I wasn't sure and did `null`, but we can probably default to `Discover` (eg. `Both`) if we have it as a type.

Reply:
We'll want to backfill, no point in setting a default - it would just be 'discover' or something, it's pretty meaningless

Reply:
There are no other dataset types for discover queries, so I'm like... meh, but I don't have a strong preference

Reply:
If all the existing records have the same value, a default would let you backfill all existing rows at once. In the future will `null` be a valid state? If not, adding a default and disallowing null would be better.

Reply:
Ah, good point Mark, I'll add a default

Reply:
Added default dataset + a separate db and state migration like our docs suggest https://github.com/getsentry/sentry/pull/71379/commits/3f02bdb7edb8de0c8c074bc3484e241254dd9009